cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -648,10 +652,13 @@ func (scaleSet *ScaleSet) TemplateNodeInfo() (*schedulerframework.NodeInfo, erro
 
 // Nodes returns a list of all nodes that belong to this node group.
 func (scaleSet *ScaleSet) Nodes() ([]cloudprovider.Instance, error) {
-	curSize, err := scaleSet.getCurSize()
-	if err != nil {
-		klog.Errorf("Failed to get current size for vmss %q: %v", scaleSet.Name, err)
-		return nil, err
+	curSize, getVMSSError := scaleSet.getCurSize()
+	if getVMSSError != nil {
+		klog.Errorf("Failed to get current size for vmss %q: %v", scaleSet.Name, getVMSSError.error)
+		if getVMSSError.notFound {
+			return nil, nil // Don't return error if VMSS not found
```

Comment:
Should we return empty array instead of `nil`?

Reply:
hm I'm not 100% sure. We don't always follow the same convention here in our codebase wrt empty vs. nil slice return values. 
fwiw this random blog I found suggests nil: https://blog.boot.dev/golang/empty-vs-nil-slices-golang/ and the code previously used nil.

Reply:
What do other providers do?

Reply:
* [azure vms pool inits a slice (and thus would return empty)](https://github.com/kubernetes/autoscaler/blob/8c71ebfe2e16262c27adef61c6285cd9c9361dc6/cluster-autoscaler/cloudprovider/azure/azure_vms_pool.go#L156)
* [aws inits a slice (and thus would return empty)](https://github.com/kubernetes/autoscaler/blob/8c71ebfe2e16262c27adef61c6285cd9c9361dc6/cluster-autoscaler/cloudprovider/aws/aws_cloud_provider.go#L369C2-L369C60)
* [gce inits a slice (and thus would return empty)](https://github.com/kubernetes/autoscaler/blob/8c71ebfe2e16262c27adef61c6285cd9c9361dc6/cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup.go#L235)

think robin's idea is better then, will refactor to do something similar